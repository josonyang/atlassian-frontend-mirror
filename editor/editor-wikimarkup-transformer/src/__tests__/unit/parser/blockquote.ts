import WikiMarkupTransformer from '../../../index';

describe('JIRA wiki markup - blockquote', () => {
	const testCases: Array<[string, string]> = [
		['[CS-607] should not need a whitespace', 'bq.this is a blockquote node'],
		['Blockquote', '{quote}This is a great quote!{quote}'],
		['Blockquote with nested codeblock', "bq.{code:javascript}pls don't put code in quotes{code}"],
		[
			'Blockquote macro with nested codeblock',
			"{quote}{code:javascript}pls don't put code in quotes{code}{quote}",
		],
		['Blockquote with nested mediaSingle', 'bq.!abc-1|width=200,height=183,alt="Hello world"!'],
		[
			'Blockquote macro with nested mediaSingle',
			'{quote}!abc-1|width=200,height=183,alt="Hello world"!{quote}',
		],
		['Blockquote with nested mediaGroup', 'bq.[^file1.txt] [^file2.txt]{quote}'],
		['Blockquote macro with nested mediaGroup', '{quote}[^file1.txt] [^file2.txt]{quote}'],
	];

	for (const [testCaseDescription, markup] of testCases) {
		it(testCaseDescription, () => {
			const transformer = new WikiMarkupTransformer();
			expect(transformer.parse(markup)).toMatchSnapshot();
		});
	}
});
