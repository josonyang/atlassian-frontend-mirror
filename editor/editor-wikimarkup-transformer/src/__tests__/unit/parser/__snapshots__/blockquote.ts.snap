// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - blockquote [CS-607] should not need a whitespace 1`] = `
{
  "content": [
    {
      "attrs": {
        "localId": null,
      },
      "content": [
        {
          "attrs": {
            "localId": null,
          },
          "content": [
            {
              "text": "this is a blockquote node",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - blockquote Blockquote 1`] = `
{
  "content": [
    {
      "attrs": {
        "localId": null,
      },
      "content": [
        {
          "attrs": {
            "localId": null,
          },
          "content": [
            {
              "text": "This is a great quote!",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - blockquote Blockquote macro with nested codeblock 1`] = `
{
  "content": [
    {
      "attrs": {
        "localId": null,
      },
      "content": [
        {
          "attrs": {
            "language": "javascript",
            "localId": null,
            "uniqueId": null,
          },
          "content": [
            {
              "text": "pls don't put code in quotes",
              "type": "text",
            },
          ],
          "type": "codeBlock",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - blockquote Blockquote macro with nested mediaGroup 1`] = `
{
  "content": [
    {
      "attrs": {
        "localId": null,
      },
      "content": [
        {
          "content": [
            {
              "attrs": {
                "__contextId": null,
                "__displayType": null,
                "__external": false,
                "__fileMimeType": null,
                "__fileName": null,
                "__fileSize": null,
                "__mediaTraceId": null,
                "alt": "",
                "collection": "",
                "height": null,
                "id": "file1.txt",
                "localId": null,
                "occurrenceKey": null,
                "type": "file",
                "url": null,
                "width": null,
              },
              "type": "media",
            },
            {
              "attrs": {
                "__contextId": null,
                "__displayType": null,
                "__external": false,
                "__fileMimeType": null,
                "__fileName": null,
                "__fileSize": null,
                "__mediaTraceId": null,
                "alt": "",
                "collection": "",
                "height": null,
                "id": "file2.txt",
                "localId": null,
                "occurrenceKey": null,
                "type": "file",
                "url": null,
                "width": null,
              },
              "type": "media",
            },
          ],
          "type": "mediaGroup",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - blockquote Blockquote macro with nested mediaSingle 1`] = `
{
  "content": [
    {
      "attrs": {
        "localId": null,
      },
      "content": [
        {
          "attrs": {
            "layout": "center",
            "localId": null,
            "width": null,
            "widthType": null,
          },
          "content": [
            {
              "attrs": {
                "__contextId": null,
                "__displayType": null,
                "__external": false,
                "__fileMimeType": null,
                "__fileName": null,
                "__fileSize": null,
                "__mediaTraceId": null,
                "alt": "Hello world",
                "collection": "",
                "height": 183,
                "id": "abc-1",
                "localId": null,
                "occurrenceKey": null,
                "type": "file",
                "url": null,
                "width": 200,
              },
              "type": "media",
            },
          ],
          "type": "mediaSingle",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - blockquote Blockquote with nested codeblock 1`] = `
{
  "content": [
    {
      "attrs": {
        "localId": null,
      },
      "content": [
        {
          "attrs": {
            "language": "javascript",
            "localId": null,
            "uniqueId": null,
          },
          "content": [
            {
              "text": "pls don't put code in quotes",
              "type": "text",
            },
          ],
          "type": "codeBlock",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - blockquote Blockquote with nested mediaGroup 1`] = `
{
  "content": [
    {
      "attrs": {
        "localId": null,
      },
      "content": [
        {
          "content": [
            {
              "attrs": {
                "__contextId": null,
                "__displayType": null,
                "__external": false,
                "__fileMimeType": null,
                "__fileName": null,
                "__fileSize": null,
                "__mediaTraceId": null,
                "alt": "",
                "collection": "",
                "height": null,
                "id": "file1.txt",
                "localId": null,
                "occurrenceKey": null,
                "type": "file",
                "url": null,
                "width": null,
              },
              "type": "media",
            },
            {
              "attrs": {
                "__contextId": null,
                "__displayType": null,
                "__external": false,
                "__fileMimeType": null,
                "__fileName": null,
                "__fileSize": null,
                "__mediaTraceId": null,
                "alt": "",
                "collection": "",
                "height": null,
                "id": "file2.txt",
                "localId": null,
                "occurrenceKey": null,
                "type": "file",
                "url": null,
                "width": null,
              },
              "type": "media",
            },
          ],
          "type": "mediaGroup",
        },
      ],
      "type": "blockquote",
    },
    {
      "attrs": {
        "localId": null,
      },
      "content": [
        {
          "attrs": {
            "localId": null,
          },
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - blockquote Blockquote with nested mediaSingle 1`] = `
{
  "content": [
    {
      "attrs": {
        "localId": null,
      },
      "content": [
        {
          "attrs": {
            "layout": "center",
            "localId": null,
            "width": null,
            "widthType": null,
          },
          "content": [
            {
              "attrs": {
                "__contextId": null,
                "__displayType": null,
                "__external": false,
                "__fileMimeType": null,
                "__fileName": null,
                "__fileSize": null,
                "__mediaTraceId": null,
                "alt": "Hello world",
                "collection": "",
                "height": 183,
                "id": "abc-1",
                "localId": null,
                "occurrenceKey": null,
                "type": "file",
                "url": null,
                "width": 200,
              },
              "type": "media",
            },
          ],
          "type": "mediaSingle",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;
